src/core/execute.c

```diff
@@ -3216,7 +3216,7 @@ static int exec_child(
                 }
 
                 fds_with_exec_fd = newa(int, n_fds + 1);
-                memcpy(fds_with_exec_fd, fds, n_fds * sizeof(int));
+                memcpy_safe(fds_with_exec_fd, fds, n_fds * sizeof(int));
```

Comment:
This one seems to fix https://github.com/systemd/systemd/issues/9738, not #9735.